### PR TITLE
Alt job titles

### PR DESCRIPTION
**Pull request by BootlegBow**
if this works first try then i'm giving myself a cookie

## About The Pull Request

adds alternate job titles

## Why It's Good For The Game

ask kirie

## Changelog
:cl:
add: alt job titles
/:cl:

### CODE REVIEW

**Comment by Kitsunemitsu** [comment]
So it's fixed?

File: code/modules/jobs/job_types/_job.dm [hunk 1]
```diff
@@ -149,8 +149,13 @@
...
 
-	if(outfit_override || outfit)
-		H.equipOutfit(outfit_override ? outfit_override : outfit, visualsOnly)
+	if(outfit && preference_source?.prefs?.alt_titles_preferences[title] && !outfit_override)//bungalow alt job titles start
+		var/outfitholder = "[outfit]/[ckey(preference_source.prefs.alt_titles_preferences[title])]"
+		if(text2path(outfitholder) || !outfitholder)
+			outfit_override = text2path(outfitholder)//bungalow end
+
+		if(outfit_override || outfit)
+			H.equipOutfit(outfit_override ? outfit_override : outfit, visualsOnly, preference_source)//bungalow edit - alt job titles
```
Comment (EgorDinamit) [inline code/modules/jobs/job_types/_job.dm:158]:
```suggestion
	if(outfit_override || outfit)
		H.equipOutfit(outfit_override ? outfit_override : outfit, visualsOnly, preference_source)//bungalow edit - alt job titles
```
Wrong indentation.

File: ModularBungalow/bungalowconfig.dm [hunk 2]
```diff
@@ -0,0 +1,3 @@
+/datum/config_entry/number/senior_timelock
+	config_entry_value = null
```
Comment (EgorDinamit) [inline ModularBungalow/bungalowconfig.dm:2]: I think you don't want to leave it as null.

**Comment by EgorDinamit** [comment]
You might also want to check these PRs to avoid other issues:
https://github.com/vlggms/tegustation/pull/86/files
https://github.com/vlggms/tegustation/pull/91/files

**Comment by EgorDinamit** [comment]
You forgot to define **truerank** variable.

File: code/datums/datacore.dm [hunk 3]
```diff
@@ -221,13 +221,20 @@
...
 			assignment = H.job
 		else
 			assignment = "Unassigned"
 
+//Bungalow edit - Alt job titles
+	trueassignment = assignment
+	if(C && C.prefs && C.prefs.alt_titles_preferences[assignment])
+		assignment = C.prefs.alt_titles_preferences[assignment]
+//Bungalow edit end
+
```
Comment (EgorDinamit) [inline code/datums/datacore.dm:237]:
```suggestion
		if(C && C.prefs && C.prefs.alt_titles_preferences[assignment])
			trueassignment = assignment
			assignment = C.prefs.alt_titles_preferences[assignment]
//Bungalow edit end

```
Hello funny indentation.

File: code/datums/datacore.dm [hunk 4]
```diff
@@ -166,7 +166,7 @@
 		var/has_department = FALSE
 		for(var/department in departments)
 			var/list/jobs = departments[department]
-			if(rank in jobs)
+			if((rank in jobs) || (truerank in jobs))//bungalow edit - alt job titles
```
Comment (EgorDinamit) [inline code/datums/datacore.dm:169]:
```suggestion
			if((rank in jobs) || (t.fields["truerank"] && (t.fields["truerank"] in jobs))) //bungalow edit - alt job titles
```

File: code/modules/client/preferences.dm [hunk 5]
```diff
@@ -923,10 +928,12 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
 				HTML += "<font color=orange>[rank]</font></td><td></td></tr>"
 				continue
+			var/rank_title_line = "[displayed_rank]" //bungalow edit-alt job titles
 			if((rank in GLOB.command_positions) || (rank == "AI"))//Bold head jobs
-				HTML += "<b><span class='dark'>[rank]</span></b>"
+				rank_title_line = "<b><a href='?_src_=prefs;preference=job;task=alt_title;job_title=[job.title]'>[rank_title_line]</a></b>"//bungalow alt job titles
 			else
-				HTML += "<span class='dark'>[rank]</span>"
+				rank_title_line = "<a href='?_src_=prefs;preference=job;task=alt_title;job_title=[job.title]'>[rank_title_line]</a>"//tegu - alt title
+				HTML += rank_title_line//tegu - alt title
```
Comment (EgorDinamit) [inline code/modules/client/preferences.dm:936]:
```suggestion
			HTML += rank_title_line//tegu - alt title
```
HELLO MISTER INDENTATION!

**Comment by Kitsunemitsu** [comment]
There we go apc destroyed mission accomplished

**Pull request merged**